deploy.sh

```diff
@@ -1,28 +1,29 @@
 #!/usr/bin/env bash
-set -e
+set -eu
 
-stack_name="$1"
-common_stack_name="${2:-pdv-matching-common-cri-api-local}"
-secret_prefix="${3:-pdv-matching-cri-api}"
+stack_name="${1:-}" && shift
 
-if [ -z "$stack_name" ]
-then
-echo "😱 stack name expected as first argument, e.g. ./deploy pdv-matching-user1"
-exit 1
+if ! [[ "$stack_name" ]]; then
+  echo "😱 Stack name expected as first argument, e.g. ./deploy.sh pdv-matching-user1"
+  exit 1
 fi
 
 sam validate -t infrastructure/template.yaml
-sam build -t infrastructure/template.yaml
+sam validate -t infrastructure/template.yaml --lint
+
+sam build -t infrastructure/template.yaml --cached --parallel
+
 sam deploy --stack-name "$stack_name" \
-   --no-fail-on-empty-changeset \
-   --no-confirm-changeset \
-   --resolve-s3 \
-   --region eu-west-2 \
-   --capabilities CAPABILITY_IAM \
-   --parameter-overrides \
-   CodeSigningEnabled=false \
-   Environment=dev \
-   AuditEventNamePrefix=/common-cri-parameters/PdvMatchingAuditEventNamePrefix \
-   CriIdentifier=/common-cri-parameters/PdvMatchingCriIdentifier \
-   CommonStackName="$common_stack_name" \
-   SecretPrefix="$secret_prefix"
+  --no-fail-on-empty-changeset \
+  --no-confirm-changeset \
+  --resolve-s3 \
+  --s3-prefix "$stack_name" \
+  --region "${AWS_REGION:-eu-west-2}" \
+  --capabilities CAPABILITY_IAM \
+  --tags \
+  cri:component=check-hmrc-api \
+  cri:stack-type=dev \
+  cri:application=Orange \
+  cri:deployment-source=manual \
+  --parameter-overrides \
+  Environment=dev
```

Comment:
@surajkumar @Connor-Johnson-DCO what do guys think about this I think you've favored using the `samconfig.toml`. I suppose the toml can be updated to have tags as well

Reply:
You can use tags with toml at the same time - but this PR doesn't change the mechanism we use to deploy so any work on toml would be a separate PR (this script doesn't use the toml file since it can be run from the root of the repo)

Reply:
Perhaps we should stick to `./deploy.sh`  if we want to benefit from this change, since peps can use sam guided and generate toml without tagging.

Reply:
The script can detect a toml file but this is again a separate piece of work since it required changing the logic